src/DSPC.sol

```diff
@@ -0,0 +1,300 @@
+        if (what == "min") {
+            require(data <= _cfgs[id].max, "DSPC/min-too-high");
+            _cfgs[id].min = uint16(data);
+        } else if (what == "max") {
+            require(data >= _cfgs[id].min, "DSPC/max-too-low");
+            _cfgs[id].max = uint16(data);
+        } else if (what == "step") {
+            _cfgs[id].step = uint16(data);
+        } else {
+            revert("DSPC/file-unrecognized-param");
+        }
+
+        emit File(id, what, data);
+    }
+
+    /// @notice Apply rate updates
+    /// @param updates Array of rate updates to apply
+    /// @dev Each update is validated against configured constraints before being applied
+    /// @dev Emits Set event after all updates are successfully applied
+    /// @dev Reverts if:
+    ///      - Empty updates array
+    ///      - Cooldown not expired
+    ///      - Rate below configured minimum
+    ///      - Rate above configured maximum
+    ///      - Rate change above configured step
+    function set(ParamChange[] calldata updates) external toll good {
+        require(updates.length > 0, "DSPC/empty-batch");
+        require(block.timestamp >= tau + toc, "DSPC/cooldown-not-expired");
+        toc = block.timestamp;
+
+        // Validate all updates in the batch
+        for (uint256 i = 0; i < updates.length; i++) {
+            bytes32 id = updates[i].id;
+            uint256 bps = updates[i].bps;
+            Cfg memory cfg = _cfgs[id];
+
+            require(bps >= cfg.min, "DSPC/below-min");
+            require(bps <= cfg.max, "DSPC/above-max");
+
+            // Check rate change is within allowed gap
+            uint256 oldBps;
+            if (id == "DSR") {
+                oldBps = conv.rtob(PotLike(pot).dsr());
+            } else if (id == "SSR") {
+                oldBps = conv.rtob(SUSDSLike(susds).ssr());
+            } else {
+                (uint256 duty,) = JugLike(jug).ilks(id);
+                oldBps = conv.rtob(duty);
+            }
+
+            uint256 delta = bps > oldBps ? bps - oldBps : oldBps - bps;
+            require(delta <= cfg.step, "DSPC/delta-above-step");
+
+            // Execute the update
+            uint256 ray = conv.btor(bps);
+            if (id == "DSR") {
+                pot.drip();
+                pot.file("dsr", ray);
+            } else if (id == "SSR") {
+                susds.drip();
+                susds.file("ssr", ray);
+            } else {
+                jug.drip(id);
+                jug.file(id, "duty", ray);
+            }
+            emit Set(id, bps);
+        }
+    }
+
+    /// @notice Calculates absolute difference between two uint256 values
+    /// @param a The first uint256 value
+    /// @param b The second uint256 value
+    /// @return delta The absolute difference between the parameters
+    function _absDiff(uint256 a, uint256 b) internal pure returns (uint256 delta) {
+        delta = a > b ? a - b : b - a;
+    }
```

Comment:
Let's just remove this.
The logic is used only once, so maybe the function is superfluous.
We can just add a comment above the ternary operator explaining what it does.
```suggestion
```